### PR TITLE
js: Add npm package for meshoptimizer

### PR DESCRIPTION
**Pull request by zeux**

### CODE REVIEW

**Review by donmccurdy — COMMENTED** [review]
While things do work without them, idiomatically you'll usually want to define entrypoints. Suggested:

```js
// index.js
const MeshoptEncoder = require('./meshopt_encoder.js');
const MeshoptDecoder = require('./meshopt_decoder.js');
module.exports = {MeshoptEncoder, MeshoptDecoder};

// index.module.js
export * from './meshopt_encoder.module.js';
export * from './meshopt_decoder.module.js';

// index.module.d.ts
export * from './meshopt_encoder.module';
export * from './meshopt_decoder.module';

// package.json
"main": "index.js",
"module": "index.module.js",
"types": "index.module.d.ts",
```

This allows the library's users to write ...

```js
import { MeshoptEncoder, MeshoptDecoder } from 'meshoptimizer';
```

... instead of ...

```js
import { MeshoptEncoder } from 'meshoptimizer/meshopt_encoder.module.js';
import { MeshoptDecoder } from 'meshoptimizer/meshopt_decoder.module.js';
```

... and makes maintaining backward-compatibility easier if file names or structure changes over time.

File: js/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		"compression",
+		"mesh"
+	],
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/zeux/meshoptimizer"
+	},
+	"files": [
+		"meshopt_*.js"
+	],
```
[the remark this answers is not in the document]
Comment (zeux) [reply js/package.json:19]: Ah right. I didn’t want to include the legacy extension modules for three/babylon (both are upstreamed). I should just delete them.

**Comment by zeux** [comment]
Re: index - it makes sense ergonomically but using this will compile and instantiate both wasm modules I believe, so it’s less efficient. Maybe still reasonable to include as an option?

**Comment by donmccurdy** [comment]
With the CommonJS entry, yes, that's a concern. The ES6 Module entrypoint won't have that issue if the user only imports one of the modules, and will shake the unused module out entirely in production (with standard build processes).

**Pull request merged**